plugin-server/src/worker/ingestion/event-pipeline/runner.ts

```diff
@@ -79,115 +41,110 @@ export class EventPipelineRunner {
+
+        const eventWithTeam = await this.runStep(populateTeamDataStep, this, event)
+        if (eventWithTeam != null) {
+            return await this.runEventPipeline(eventWithTeam)
+        }
+
         this.hub.statsd?.increment('kafka_queue.single_event.processed_and_ingested')
-        return result
     }
 
-    async runEventPipeline(event: PluginEvent): Promise<EventPipelineResult> {
+    async runEventPipeline(event: PluginEvent) {
         this.hub.statsd?.increment('kafka_queue.event_pipeline.start', { pipeline: 'event' })
-        const result = await this.runPipeline('emitToBufferStep', event)
+        await this.runEventPipelineSteps(event)
         this.hub.statsd?.increment('kafka_queue.single_event.processed_and_ingested')
-        return result
     }
 
-    async runBufferEventPipeline(event: PluginEvent): Promise<EventPipelineResult> {
+    async runEventPipelineSteps(event: PluginEvent) {
+        const personContainer = new LazyPersonContainer(event.team_id, event.distinct_id, this.hub)
+        const bufferResultEvent = await this.runStep(emitToBufferStep, this, event, personContainer)
+        if (bufferResultEvent != null) {
+            const processedEvent = await this.runStep(pluginsProcessEventStep, this, bufferResultEvent)
+
+            if (processedEvent) {
+                const [normalizedEvent, newPersonContainer] = await this.runStep(
+                    processPersonsStep,
+                    this,
+                    processedEvent,
+                    personContainer
+                )
+
+                const preparedEvent = await this.runStep(prepareEventStep, this, normalizedEvent)
+                if (preparedEvent != null) {
+                    return await this.runStep(createEventStep, this, preparedEvent, newPersonContainer)
+                }
+            }
+        }
+    }
+
+    async runBufferEventPipeline(event: PluginEvent) {
         this.hub.statsd?.increment('kafka_queue.event_pipeline.start', { pipeline: 'buffer' })
         const personContainer = new LazyPersonContainer(event.team_id, event.distinct_id, this.hub)
         // We fetch person and check for existence for metrics for buffer efficiency
         const didPersonExistAtStart = !!(await personContainer.get())
 
-        const result = await this.runPipeline('pluginsProcessEventStep', event, personContainer)
+        const processedEvent = await this.runStep(pluginsProcessEventStep, this, event)
+
+        if (processedEvent) {
+            const [normalizedEvent, newPersonContainer] = await this.runStep(
+                processPersonsStep,
+                this,
+                processedEvent,
+                personContainer
+            )
 
+            const preparedEvent = await this.runStep(prepareEventStep, this, normalizedEvent)
+            if (preparedEvent != null) {
+                return await this.runStep(createEventStep, this, preparedEvent, newPersonContainer)
+            }
+        }
```

Comment:
I'd pull this into a function as it's dupe between this and above `runEventPipelineStepsAfterBuffer` or whatever you'd want to call it